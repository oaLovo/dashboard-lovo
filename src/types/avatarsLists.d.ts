export type ListDataProps = {
  avatar?: string
  name: string
  subtitle: string
}

export interface AvatarsListsProps {
  listData: ListDataProps[]
}
