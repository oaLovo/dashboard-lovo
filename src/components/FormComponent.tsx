import styled from 'styled-components'
import { FormComponentProps } from '@/types'
import { StyledButton, StyledInput, StyledP } from '@/components'
import { pxToRem } from '@/utils'

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${pxToRem(16)};
`

function FormComponent(props: FormComponentProps) {
  const { inputs, buttons, message } = props
  return (
    <StyledForm>
      {inputs.map((inputProps, index) => (
        <StyledInput key={index} {...inputProps} />
      ))}
      {buttons.map((buttonProps, index) => (
        <StyledButton key={index} {...buttonProps} />
      ))}
      {message && (
        <StyledP className={message.type === 'error' ? 'error' : 'success'}>
          {message.msg}
        </StyledP>
      )}
    </StyledForm>
  )
}

export default FormComponent
