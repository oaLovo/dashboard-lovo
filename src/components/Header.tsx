import styled from 'styled-components'
import { Link } from 'react-router-dom'

import { Logo } from '@/components'

// MUI
import Avatar from '@mui/material/Avatar'
import Box from '@mui/material/Box'
import Container from '@mui/material/Container'

// UTILS
import { pxToRem } from '@/utils'

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`

function Header() {
  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={30} width={73} />
          </Link>
          <Link to="/perfil">
            <Avatar
              alt="DNC Avatar"
              src="/dnc-avatar.jpg"
              sx={{ width: pxToRem(40), height: pxToRem(40) }}
            />
          </Link>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
