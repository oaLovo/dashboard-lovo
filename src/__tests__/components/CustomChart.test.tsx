import { BrowserRouter as Router } from 'react-router-dom'
import { CustomChart } from '@/components'
import { CustomChartProps } from '@/types'
import { render } from '@testing-library/react'
import { Theme } from '@/types'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@/resources/themesList'

const barProps: CustomChartProps = {
  data: [10, 20, 30],
  labels: ['January', 'February', 'March'],
  type: 'bar',
}

const lineProps: CustomChartProps = {
  data: [10, 20, 30],
  labels: ['January', 'February', 'March'],
  type: 'line',
}

describe('Custom chart Component', () => {
  const renderComponent = (props: CustomChartProps, theme: Theme) =>
    render(
      <ThemeProvider theme={theme}>
        <Router>
          <CustomChart {...props} />
        </Router>
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot - bar chart', () => {
        const { container } = renderComponent(barProps, theme)
        expect(container).toMatchSnapshot()
      })
      it('should match the snapshot - line chart', () => {
        const { container } = renderComponent(lineProps, theme)
        expect(container).toMatchSnapshot()
      })
    })
  })
})
