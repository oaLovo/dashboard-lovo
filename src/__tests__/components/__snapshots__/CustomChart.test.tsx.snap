// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom chart Component Dark Theme should match the snapshot - bar chart 1`] = `
<div>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</div>
`;

exports[`Custom chart Component Dark Theme should match the snapshot - line chart 1`] = `
<div>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</div>
`;

exports[`Custom chart Component Light Theme should match the snapshot - bar chart 1`] = `
<div>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</div>
`;

exports[`Custom chart Component Light Theme should match the snapshot - line chart 1`] = `
<div>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</div>
`;
